SparkPost/2web2ui
# AC-1911 - Fix Safari SSO redirect issue

AC-1911 - Fix Safari SSO redirect issue

Note - more discussion regarding this - https://msys.slack.com/archives/GKXTKHDSP/p1616099635000400

### What Changed

- SameSite for authCookie is set to lax for safari and kept as 'strict' for other browsers
- Installed is_js for browser detection
- Made SSOPage to be a functional component and updated tests.
Safari
![Screen Shot 2021-03-19 at 10 22 32 AM](https://user-images.githubusercontent.com/4179337/111800521-6b3a6280-88a2-11eb-80ca-f308286037ca.png)
Chrome
![Screen Shot 2021-03-19 at 10 52 55 AM](https://user-images.githubusercontent.com/4179337/111800561-75f4f780-88a2-11eb-9fdc-3ba0b19ebb81.png)


### How To Test

- Point openresty to uat.
- please let me know when you test this I can run accusers locally(could be an issue with aws persimissions to run this) and also share with you my okta creds to test this.
- Check if you can reproduce the bug in safari

> **Steps to Reproduce**
> Visit https://app.sparkpost.test:3100 with Safari
> Click "Single Sign-On" link
> Authenticate with SSO
> Close browser tab
> Open new tab and visit https://app.sparkpost.test:3100

- Check the value for samesite for auth cookie in safari and chrome.

    

### To Do
